server/repository/src/db_diesel/currency.rs

```diff
@@ -0,0 +1,107 @@
+use super::{
+    currency_row::{currency, currency::dsl as currency_dsl},
+    CurrencyRow, DBType, StorageConnection,
+};
+use diesel::prelude::*;
+
+use crate::{
+    diesel_macros::{apply_equal_filter, apply_sort_no_case},
+    repository_error::RepositoryError,
+};
+
+use crate::{EqualFilter, Pagination, Sort};
+
+#[derive(PartialEq, Debug, Clone)]
+pub struct Currency {
+    pub currency_row: CurrencyRow,
+}
+
+#[derive(Clone, PartialEq, Debug)]
+pub struct CurrencyFilter {
+    pub id: Option<EqualFilter<String>>,
+}
+
+#[derive(PartialEq, Debug)]
+pub enum CurrencySortField {
+    Id,
+    CurrencyCode,
+}
+
+pub type CurrencySort = Sort<CurrencySortField>;
+
+pub struct CurrencyRepository<'a> {
+    connection: &'a StorageConnection,
+}
+
+impl<'a> CurrencyRepository<'a> {
+    pub fn new(connection: &'a StorageConnection) -> Self {
+        CurrencyRepository { connection }
+    }
+
+    pub fn count(&self, filter: Option<CurrencyFilter>) -> Result<i64, RepositoryError> {
+        let query = create_filtered_query(filter);
+        Ok(query.count().get_result(&self.connection.connection)?)
+    }
+
+    pub fn query_by_filter(
+        &self,
+        filter: CurrencyFilter,
+    ) -> Result<Vec<Currency>, RepositoryError> {
+        self.query(Pagination::all(), Some(filter), None)
+    }
+
+    pub fn query(
+        &self,
+        pagination: Pagination,
```

Comment:
🤔  maybe pagination isn't required for this one? there won't be very many currencies, and when we want to get them, we would want all available currencies